app/enssubdomain/views.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+"""
+    Copyright (C) 2018 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import datetime
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.http import JsonResponse
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+from django.views.decorators.csrf import csrf_exempt
+from gas.utils import recommend_min_gas_price_to_confirm_in_time
+
+from dashboard.models import Profile
+from dashboard.views import w3
+from ens import ENS
+from eth_account.messages import defunct_hash_message
+from web3 import HTTPProvider, Web3
+from enssubdomain.web3py.web3.middleware.signing import construct_sign_and_send_raw_middleware
+
+from .models import ENSSubdomainRegistration
+
+ns = ENS.fromWeb3(w3)
+w3 = Web3(HTTPProvider(settings.WEB3_HTTP_PROVIDER))
+ENS_MAINNET_ADDR = '0x314159265dD8dbb310642f98f50C066173C1259b'
```

Comment:
wondering why you need this. The ens implementation in web3 already  has this hardcoded.

Reply:
good call.. removing it now